dbms/src/Server/MetricsPrometheus.cpp

```diff
@@ -181,19 +195,12 @@ MetricsPrometheus::MetricsPrometheus(
             auto host = metrics_addr.substr(0, pos);
             auto port = metrics_addr.substr(pos + 1, metrics_addr.size());
 
-            auto service_addr = conf.getString("flash.service_addr");
-            std::string job_name = service_addr;
-            std::replace(job_name.begin(), job_name.end(), ':', '_');
-            std::replace(job_name.begin(), job_name.end(), '.', '_');
-            job_name = "tiflash_" + job_name;
-
-            char hostname[1024];
-            ::gethostname(hostname, sizeof(hostname));
-
-            gateway = std::make_shared<prometheus::Gateway>(host, port, job_name, prometheus::Gateway::GetInstanceLabel(hostname));
+            const String & job_name = "tiflash";
+            const auto & labels = prometheus::Gateway::GetInstanceLabel(conf.getString("flash.service_addr", getHostName()));
```

Comment:
does it means that it will use as label getHostName as value if "flash.service_addr" is null?

Reply:
the `flash.service_addr` won't be null, isn't it? what if service_addr is sth like "0.0.0.0:1234"

Reply:
> does it means that it will use as label getHostName as value if "flash.service_addr" is null?
yes

> what if service_addr is sth like "0.0.0.0:1234"
good point!
In testing locally, the result of `getHostName` is `P99`, so `flash.service_addr` is used here.
Looks like we need to use `getHostName` when service_addr are `127.0.0.1` and `0.0.0.0`

Reply:
> > does it means that it will use as label getHostName as value if "flash.service_addr" is null?
> 
> yes
> 
> > what if service_addr is sth like "0.0.0.0:1234"
> 
> good point! In testing locally, the result of `getHostName` is `h99`, so `flash.service_addr` is used here. Looks like we need to use `getHostName` when service_addr are `127.0.0.1` and `0.0.0.0`

updated.